src/Data/Poly/Internal/Dense.hs

```diff
@@ -159,6 +167,25 @@ instance (Eq a, Num a, G.Vector v a) => Num (Poly v a) where
   {-# INLINE fromInteger #-}
   {-# INLINE (*) #-}
 
+quotPoly :: forall a v. (Eq a, Integral a, G.Vector v a) => Poly v a -> Poly v a -> Poly v a
+quotPoly a b = fst $ quotRemPoly a b
+
+quotRemPoly :: forall a v. (Eq a, Integral a, G.Vector v a) => Poly v a -> Poly v a -> (Poly v a, Poly v a)
```

Comment:
Is it possible to state a property defining `quotRemPoly` uniquely? E. g., imagine `(q, r) = quotRemPoly X 2`. In which sense the remainder `r` (probably, equal to `X` in your implementation) is "smaller" than the divisor `2`?

The thing is that it is impossible to define division with remainder given `Integral a` only. That is why `Data.Poly.Internal.Dense.Field` requires at least `Field a`.

What do you use this function for?